tests/ui/traits/next-solver/coroutine-higher-ranked-rigid.rs

```diff
@@ -0,0 +1,33 @@
+//@ compile-flags: -Znext-solver
+//@ edition: 2024
+//@ check-pass
+
+// Regression test for <https://github.com/rust-lang/trait-system-refactor-initiative/issues/177>.
+// Coroutines erase all free lifetimes from their interior types. In the future in `from_request`,
+// that means the `'r` lifetime in the type `<T as FromRequest<'r>>::Assoc`, which is in the
+// `dyn Future` type it's awaiting. Normalizing this associated type, with its free lifetimes
```

Comment:
wordiong "that means the `'r` lifetime..." what is the `'r` lifetime doing. "Is getting replaced by a higher ranked variable?"